jcli/src/jcli_lib/vote/tally/decrypt_tally.rs

```diff
@@ -76,7 +79,9 @@ impl TallyVotePlanWithAllShares {
                 }
             }
         }
-        let table = chain_vote::TallyOptimizationTable::generate(max_stake);
+        let table = chain_vote::TallyOptimizationTable::generate(
+            NonZeroU64::new(max_stake).ok_or(Error::ZeroStakeWhileDecryptingTally)?,
+        );
```

Comment:
The aim with the chain-libs change is that consumers of the API don't need to worry about setting up a table, instead there's a function `batch_decrypt(&[ValidatedTally]) -> Result<Vec<Tally>>` which sets up and reuses a table for all the tallies. It also handles the zero case, so that error variant is probably not needed.

`chain-libs/chain-impl-mockchain/src/testing/builders/vote.rs` has a `decrypt_tally` function like this that's changed pretty significantly, should be able to just copy/paste from there

Reply:
I'm going to take on that.